edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Http/RegistryApiClient.cs

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.Http
+{
+    using System;
+    using System.Globalization;
+    using System.Net;
+    using System.Net.Http;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    public class RegistryApiClient : IRegistryApiClient
+    {
+        internal const string MediaType = "application/json";
+        const string PutModuleOnBehalfOfUriTemplate = "devices/{0}/modules/$edgeHub/putModuleOnBehalfOf?api-version={1}";
```

Comment:
Nit: make the moduleId a parameter as well, and sub it with the EdgeHub module Id.